### ISSUE TITLE
Failed to compile packages dea_next & cloud_controller_ng on cf/215

### ISSUE THREAD

**Issue by hashneo** [cloudfoundry/cf-release#767]
Hi, I'm getting this error on deploying CF 215, deploying to Openstack.

The stemcell i'm using is:
bosh-stemcell-3042-openstack-kvm-ubuntu-trusty-go_agent-raw.tgz

I have updated everything to the latest version, including microbosh.

```
   Failed compiling packages > dea_next/7012a60317829571dbe3f3659a16f0d8e3ae136e: Action Failed get_task: Task da7756fa-c0e0-4007-5727-d95730b76735 result: Compiling package dea_next: Running packaging script: Command exited with 7; Stdout: installing your bundle as root will break this application for all non-root
users on this machine.
Could not find rake-10.0.3 in any of the sources
, Stderr: + cp -a dea_next/Gemfile dea_next/Gemfile.lock dea_next/bin dea_next/buildpacks dea_next/go dea_next/lib /var/vcap/packages/dea_next
+ cd /var/vcap/packages/dea_next
+ /var/vcap/packages/ruby-2.1.6/bin/bundle install --local --deployment --without=development test
 (00:00:11)
   Failed compiling packages > cloud_controller_ng/af5e23652500f11f7f3a3d884950dd23b75c7e56: Action Failed get_task: Task aa7c7c12-c9de-4525-5443-0d4d8ec0d3b7 result: Compiling package cloud_controller_ng: Running packaging script: Command exited with 7; Stdout: installing your bundle as root will break this application for all non-root
users on this machine.
Could not find rake-10.4.2 in any of the sources
, Stderr: + cp -a cloud_controller_ng packaging /var/vcap/packages/cloud_controller_ng
+ cd /var/vcap/packages/cloud_controller_ng/cloud_controller_ng
+ bundle_cmd=/var/vcap/packages/ruby-2.1.6/bin/bundle
+ mysqlclient_dir=/var/vcap/packages/mysqlclient
+ libpq_dir=/var/vcap/packages/libpq
+ /var/vcap/packages/ruby-2.1.6/bin/bundle config build.mysql2 --with-mysql-dir=/var/vcap/packages/mysqlclient --with-mysql-include=/var/vcap/packages/mysqlclient/include/mysql
+ /var/vcap/packages/ruby-2.1.6/bin/bundle config build.pg --with-pg-lib=/var/vcap/packages/libpq/lib --with-pg-include=/var/vcap/packages/libpq/include
+ /var/vcap/packages/ruby-2.1.6/bin/bundle install --local --deployment --without development test
 (00:00:06)
     Done compiling packages > warden/5f0154e5ba4baec289e1a05fe81ee1b9a5dc23e1 (00:00:22)

Error 450001: Action Failed get_task: Task da7756fa-c0e0-4007-5727-d95730b76735 result: Compiling package dea_next: Running packaging script: Command exited with 7; Stdout: installing your bundle as root will break this application for all non-root
users on this machine.
Could not find rake-10.0.3 in any of the sources
, Stderr: + cp -a dea_next/Gemfile dea_next/Gemfile.lock dea_next/bin dea_next/buildpacks dea_next/go dea_next/lib /var/vcap/packages/dea_next
+ cd /var/vcap/packages/dea_next
+ /var/vcap/packages/ruby-2.1.6/bin/bundle install --local --deployment --without=development test
```

**Comment by jpalermo** [comment]
Are you running bosh create release yourself or uploading the final release manifest for 215?

**Comment by hashneo** [comment]
Creating the release myself.

**Comment by jpalermo** [comment]
Try updating your version of bundler and creating the release again.

It's possible that bosh won't recreate the packages though, because it won't detect any changes. So you might need to delete the manifest (probably under dev_releases) for the release you created. You might even need to delete the .dev_builds/packages folder, or worst case, all the files under ~/.bosh/cache

When you do the create release, you should be able to tell if it is recreating the dea and cloud controller packages.

**Comment by hashneo** [comment]
But its failing on the compilation machines created by bosh, not the local machine?

**Comment by jpalermo** [comment]
Yeah, with older bundler versions, the pre-packaging doesn't package everything up correctly, so when it goes to compile it, it can't find the gems.

**Comment by hashneo** [comment]
Ok, I did a full rebuild with a new clone of the cf-release git repository and it works now. guess it was hung up somewhere.

**Comment by jpalermo** [comment]
Yeah, during pre-packaging we package the gems needed for the ruby apps. Previously we just called `bundle package --all`. Recently we changed this to `bundle package --all --no-install` which has the benefit of not installing the gems locally before packaging them. The local installation would often cause problems. Unfortunately if you are using a version of bundler that doesn't include the `--no-install` option, you end up in the state you were in.

Unfortunately we have not yet found a good way to detect this during pre-packaging, warn the user, and fail.
